arangod/RestServer/DatabaseFeature.cpp

```diff
@@ -586,9 +587,19 @@ void DatabaseFeature::recoveryDone() {
 
   // '_pendingRecoveryCallbacks' will not change because
   // !StorageEngine.inRecovery()
+  // It's single active thread before recovery done,
+  // so we could use general purpose thread pool for this
+  std::vector<futures::Future<Result>> futures;
+  futures.reserve(_pendingRecoveryCallbacks.size());
   for (auto& entry : _pendingRecoveryCallbacks) {
-    auto result = entry();
-
+    futures.emplace_back(SchedulerFeature::SCHEDULER->queueWithFuture(
```

Comment:
I don't know exactly what's in a recovery callback object, but as we are running them concurrently now, are they all fully thread-safe?

Reply:
For now yes, they're thread-safe. We also run sanitizers.
In future, if someone decide to add new postrecovery callback he will need to account this.
So doxygen was changed.